### PR TITLE
Add support for managing authentication in ara_frontend_nginx

### PR DESCRIPTION
**Pull request by dmsimard**
This includes drive-by fixes but it's mostly about adding support for managing external authentication via a .htpasswd file in nginx as per the documentation: https://ara.readthedocs.io/en/latest/api-security.html#authentication-and-user-management

The PR also includes a new integration test which covers a large gap that we had in testing -- it will set up gunicorn with nginx in front managing the authentication.

### CODE REVIEW

File: tests/test_tasks.yaml
Comment (dmsimard) [inline tests/test_tasks.yaml:89]: All jobs are failing with the callback returning HTTP 401 unauthorized errors, maybe these are being interpreted and used with empty values. Will check.

File: roles/ara_frontend_nginx/tasks/main.yaml
Comment (Thulium-Drake) [inline roles/ara_frontend_nginx/tasks/main.yaml:48]: Don't these need {{ }} ? :-)

**Review by Thulium-Drake — CHANGES_REQUESTED** [review]
Looks good!

File: roles/ara_frontend_nginx/tasks/main.yaml
Comment (dmsimard) [reply roles/ara_frontend_nginx/tasks/main.yaml:48]: good catch, I wrote this too late last night :p

**Comment by Thulium-Drake** [comment]
We're missing a python dependency for htpasswd:

```
ModuleNotFoundError: No module named 'passlib'
```

https://dashboard.zuul.ansible.com/t/ansible/build/06548ac0cdca467681c14a7e557704ed

**Comment by dmsimard** [comment]
> We're missing a python dependency for htpasswd:
> 
> ```
> ModuleNotFoundError: No module named 'passlib'
> ```
> 
> https://dashboard.zuul.ansible.com/t/ansible/build/06548ac0cdca467681c14a7e557704ed

You're too fast :laughing:

I think https://github.com/ansible-community/ara-collection/pull/38/commits/f24ec00f27487e5050198c3f11b993ee62fb2cca should fix it.

**Comment by dmsimard** [comment]
:tada: IT WORKED \o/

I'll clean it up later and we can re-review before merging.

File: roles/ara_frontend_nginx/templates/ara-api.conf.j2
Comment (dmsimard) [inline roles/ara_frontend_nginx/templates/ara-api.conf.j2:15]: This block needs fixing:

```
    server_name 127.0.0.1;
        auth_basic "Privileged access";
    # A .htpasswd file
    auth_basic_user_file /etc/nginx/.htpasswd;

```

**Comment by dmsimard** [comment]
@Thulium-Drake would you like to do a last review ?

Cleaned up/squashed and rebased stuff and it should be good to go if Zuul is happy with it.

File: roles/ara_api/tasks/install/distribution.yaml
Comment (Thulium-Drake) [inline roles/ara_api/tasks/install/distribution.yaml:31]: Debian uses the same name for this package, so it might not be required to make it a variable :-)

**Review by Thulium-Drake — APPROVED** [review]
LGTM

**Pull request merged**
